app/src/main/java/com/epfl/drawyourpath/userProfile/UserModel.kt

```diff
@@ -0,0 +1,354 @@
+package com.epfl.drawyourpath.userProfile
+
+import android.graphics.Color
+import android.text.TextUtils
+import com.epfl.drawyourpath.database.Database
+import java.time.LocalDate
+import java.util.concurrent.TimeUnit
+
+class UserModel {
+    //the userId is given by the authentication part and is unique to the user
+    private val userId: String
+    //the userName is chosen by the user and can be modify
+    private var username: String
+    //the email is given at the beginning by the authentication part and can be modify
+    private var emailAddress: String
+    //the firstname can't be modify after initialization
+    private val firstname: String
+    //the surname can't be modify after initialization
+    private val surname: String
+    //the date of birth can't be modify after initialization
+    private val dateOfBirth: LocalDate
+    //the distance goal is initialize at the profile creation and can be modify
+    private var distanceGoal: Double
+    //the activity time goal is initialize at the profile creation and can be modify
+    private var activityTimeGoal: Double
+    //the number of path goal is initialize at the profile creation and can be modify
+    private var nbOfPathsGoal: Int
+
+    //database where the user is store online
+    private var database: Database
+    //friend list
+    private var friendsList: HashMap<String, String> //(username, userId)
+
+    /**
+     * THis constructor will create a new user based on the user model of the app
+     * @param userId is given by the authentification part and is unique to the user (used to access the user profile on the database)
+     * @param username is chosen during the profile create and each user has  unique username and it can be change later(if available in the database = not taken by another user)
+     * @param firstname must respect the name convention of the app (name or name-name)
+     * @param surname must respect the name convention of the app (name or name-name)
+     * @param dateOfBirth the user be aged between 10 and 100 years old
+     * @param distanceGoal init at the user profile creation and can be modify after(daily goal)
+     * @param activityTimeGoal init at the user profile creation and can be modify after(daily goal)
+     * @param nbOfPathsGoal init at the user profile creation and can be modify after(daily goal)
+     * @throws an error if the inputs are incorrect
+     */
+    constructor(userId: String, username: String, emailAddress: String, firstname: String, surname: String, dateOfBirth: LocalDate, distanceGoal: Double, activityTimeGoal: Double, nbOfPathsGoal: Int, database: Database){
+        this.database = database
```

Comment:
The class already has setters that handle invalid cases. We should call them instead of handling errors twice.